packages/router/src/grpc/view-protocol-server/broadcast-transaction.test.ts

```diff
@@ -0,0 +1,483 @@
+import { beforeEach, describe, expect, Mock, test, vi } from 'vitest';
+import {
+  BroadcastTransactionRequest,
+  BroadcastTransactionResponse,
+  TransactionInfo,
+} from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/view/v1alpha1/view_pb';
+import { createContextValues, createHandlerContext, HandlerContext } from '@connectrpc/connect';
+import { ViewProtocolService } from '@buf/penumbra-zone_penumbra.connectrpc_es/penumbra/view/v1alpha1/view_connect';
+import { servicesCtx } from '../../ctx';
+import { Transaction } from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/core/transaction/v1alpha1/transaction_pb';
+import { broadcastTransaction } from './broadcast-transaction';
+import type { Services } from '@penumbra-zone/services';
+import { TransactionId } from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/core/txhash/v1alpha1/txhash_pb';
+
+const mockSha256 = vi.hoisted(() => vi.fn());
+vi.mock('@penumbra-zone/crypto-web', () => ({
+  sha256Hash: mockSha256,
+}));
+
+interface IndexedDbMock {
+  subscribe: (table: string) => Partial<AsyncIterable<Mock>>;
+}
+interface TendermintMock {
+  broadcastTx: Mock;
+}
+interface MockServices {
+  getWalletServices: Mock<[], Promise<{ indexedDb: IndexedDbMock }>>;
+  querier: {
+    tendermint: TendermintMock;
+  };
+}
```

Comment:
These feel like they should live in a shared place. Maybe a `test-utils.ts` file?